editor/components/block-list/invalid-block-warning.js

```diff
@@ -20,54 +20,32 @@ import {
 import { replaceBlock } from '../../store/actions';
 import Warning from '../warning';
 
-function InvalidBlockWarning( { block, attemptFixParagraph, ignoreInvalid, switchToBlockType } ) {
-	const htmlBlockName = 'core/html';
-	const defaultBlockType = getBlockType( getUnknownTypeHandlerName() );
-	const htmlBlockType = getBlockType( htmlBlockName );
-	const switchTo = ( blockType ) => () => switchToBlockType( blockType );
+function InvalidBlockWarning( { block, onReplace } ) {
+	const hasHTMLBlock = !! getBlockType( 'core/html' );
+
+	const convertToHTML = () => {
+		onReplace( block.uid, createBlock( 'core/html', {
+			content: block.originalContent,
+		} ) );
+	};
+
+	const convertToBlocks = () => {
+		onReplace( block.uid, rawHandler( {
+			HTML: block.originalContent,
+			mode: 'BLOCKS',
+		} ) );
+	};
 
 	return (
 		<Warning>
-			<p>{ defaultBlockType && htmlBlockType && sprintf( __(
-				'This block appears to have been modified externally. ' +
-				'Overwrite the changes or Convert to %s or %s to keep ' +
-				'your changes.'
-			), defaultBlockType.title, htmlBlockType.title ) }</p>
+			<p>{ __( 'This block appears to have been modified externally.' ) }</p>
```

Comment:
Note: I removed information here that appears in the buttons. No need to repeat I think.